stl/inc/memory

```diff
@@ -201,52 +322,156 @@ template <class _InIt, class _Diff, class _NoThrowFwdIt>
 pair<_InIt, _NoThrowFwdIt> uninitialized_move_n(_InIt _First, const _Diff _Count_raw, _NoThrowFwdIt _Dest) {
     // move [_First, _First + _Count) to [_Dest, ...)
     _Algorithm_int_t<_Diff> _Count = _Count_raw;
-    if (0 < _Count) {
-        auto _UFirst = _Get_unwrapped_n(_First, _Count);
-        auto _UDest  = _Get_unwrapped_n(_Dest, _Count);
-        if constexpr (_Ptr_move_cat<decltype(_UFirst), decltype(_UDest)>::_Really_trivial) {
-            _UDest = _Copy_memmove(_UFirst, _UFirst + _Count, _UDest);
-            _UFirst += _Count;
-        } else {
-            _Uninitialized_backout<decltype(_UDest)> _Backout{_UDest};
-            for (; 0 < _Count; --_Count, (void) ++_UFirst) {
-                _Backout._Emplace_back(_STD move(*_UFirst));
-            }
+    if (_Count <= 0) {
+        return {_First, _Dest};
+    }
+
+    auto _UFirst = _Get_unwrapped_n(_First, _Count);
+    auto _UDest  = _Get_unwrapped_n(_Dest, _Count);
+    if constexpr (_Ptr_move_cat<decltype(_UFirst), decltype(_UDest)>::_Really_trivial) {
+        _UDest = _Copy_memmove(_UFirst, _UFirst + _Count, _UDest);
+        _UFirst += _Count;
+    } else {
+        _Uninitialized_backout<decltype(_UDest)> _Backout{_UDest};
 
-            _UDest = _Backout._Release();
+        for (; _Count > 0; --_Count, (void) ++_UFirst) {
+            _Backout._Emplace_back(_STD move(*_UFirst));
         }
 
-        _Seek_wrapped(_Dest, _UDest);
-        _Seek_wrapped(_First, _UFirst);
+        _UDest = _Backout._Release();
     }
 
+    _Seek_wrapped(_Dest, _UDest);
+    _Seek_wrapped(_First, _UFirst);
     return {_First, _Dest};
 }
 #endif // _HAS_CXX17
 
+
+#ifdef __cpp_lib_concepts
+namespace ranges {
+    // ALIAS TEMPLATE uninitialized_move_n_result
+    template <class _In, class _Out>
+    using uninitialized_move_n_result = in_out_result<_In, _Out>;
+
+    // VARIABLE ranges::uninitialized_move_n
+    class _Uninitialized_move_n_fn : private _Not_quite_object {
+    public:
+        using _Not_quite_object::_Not_quite_object;
+
+        // clang-format off
+        template <input_iterator _It, _No_throw_forward_iterator _Out, _No_throw_sentinel_for<_Out> _OSe>
+            requires constructible_from<iter_value_t<_Out>, iter_rvalue_reference_t<_It>>
+        uninitialized_move_n_result<_It, _Out> operator()(
+            _It _First1, iter_difference_t<_It> _Count, _Out _First2, _OSe _Last2) const {
+            // clang-format on
+            if (_Count <= 0) {
+                return {_STD move(_First1), _STD move(_First2)};
+            }
+
+            _Adl_verify_range(_First2, _Last2);
+            auto _IFirst      = _Get_unwrapped_n(_STD move(_First1), _Count);
+            auto _OFirst      = _Get_unwrapped(_STD move(_First2));
+            const auto _OLast = _Get_unwrapped(_STD move(_Last2));
+            if constexpr (_Ptr_move_cat<_It, _Out>::_Really_trivial) {
+                _OFirst = _Copy_memmove_common(_IFirst, _IFirst + _Count, _OFirst, _OLast);

```

Comment:
Ditto false pattern - `ranges::uninitialized_move_n` should use `_Copy_memcpy_common` here.

Reply:
Ditto removed